lisa/runner.py

```diff
@@ -322,12 +327,14 @@ def _start_loop(self) -> None:
                         # previous runner firstly in next run.
                         break
 
-                while (
-                    len(remaining_runners) < self._max_concurrency and has_more_runner
-                ):
-                    # Fetch runners, if runner count is smaller than concurrency
-                    # count. It makes sure all concurrency can run.
-                    try:
-                        remaining_runners.append(next(runner_iterator))
-                    except StopIteration:
-                        has_more_runner = False
+                if has_idle_worker:
```

Comment:
It looks not like we discussed. The logic is not change a lot with current one.

Reply:
This is the algo we discussed. Does the current change miss any case ?

![MicrosoftTeams-image (2)](https://user-images.githubusercontent.com/10995909/134437153-78eff31f-5043-438e-b809-ab07eb1e236b.png)

Reply:
The loop for new runner should be in the while #2.